app/Http/Controllers/Admin/ActionCrudController.php

```diff
@@ -49,17 +54,90 @@ public function setup()
      */
     protected function setupListOperation()
     {
-        $this->crud->addFilter([ 
-            'type'  => 'simple',
-            'name'  => 'team_id',
-            'label' => 'Team'
-          ],
-          false,
-          function() { 
-            $teams = backpack_user()->teams;
-            
-            $this->crud->addClause('whereIn', 'team_id', $teams); 
-       
+        $this->crud->addFilter([
+            'name'  => 'teams',
+            'type'  => 'select2_multiple',
+            'label' => 'Teams'
+            ], function() {
+             
+                return Team::get()->pluck('name', 'id')->toArray();;
+
+            }, function($values) { // if the filter is active
+
+                $this->crud->addClause('whereIn', 'team_id',json_decode($values) );
+        });
+
+        $this->crud->addFilter([
+            'name'  => 'outputs',
+            'type'  => 'select2_multiple',
+            'label' => 'Ouputs'
+            ], function() {
+             
+                return Output::get()->pluck('name', 'id')->toArray();
+
+            }, function($values) { // if the filter is active
+                foreach (json_decode($values) as $key => $value) {
+                $this->crud->query = $this->crud->query->whereHas('subactivities', function ($query) use ($value) {
+                    $this->crud->query->whereHas('activities', function ($query) use ($value) {
+
+                        $query->where('output_id', $value);
+                    });
+
```

Comment:
Looking at the Action model, there is actually both a `subactivities()` and an `activities()` relationship - powered by seperate link tables in the database. So this can actually be simplified - see my upcoming commit.

Reply:
I don't sure how I can simplify the filter

Reply:
I've already done that - as I said, see my added commits 👍